### PR TITLE
Generators default values

### PR DESCRIPTION
**Pull request by Profpatsch**
better default values & documentation; see commit messages.

cc @aszlig

### CODE REVIEW

File: lib/generators.nix [hunk 1]
```diff
@@ -20,6 +26,32 @@ in
...
+   * The builtin `toString` function has some strange defaults,
+   * suitable for bash scripts but not much else.
+   */
+  mkValueStringDefault = {}: v: with builtins;
+    let err = t: v: abort
+          ("generators.mkValueStringDefault: " +
+           "${t} not supported: ${toPretty {} v}");
+    in   if isInt      v then toString v
+    # we default to not quoting strings
+    else if isString   v then ''${v}''
```
Comment (aszlig) [inline lib/generators.nix]: If `v` is already a string, why antiquote it?

File: doc/functions.xml [hunk 2]
```diff
@@ -221,16 +221,68 @@
 
   <para>
     All generators follow a similar call interface: <code>generatorName
-    configFunctions data</code>, where <literal>configFunctions</literal> is a
-    set of user-defined functions that format variable parts of the content.
+    { configFunctions } data</code>, where <literal>configFunctions</literal> is
```
Comment (aszlig) [inline doc/functions.xml]: Really? As I can see this doesn't seem to be the case, it's still like `toINI { mkSectionName = ... }` and not `toINI { configFunctions.mkSectionName = ... }`, did I miss something? And if yes, then it's a backwards-incompatibility and should have a release notes entry.

**Review by aszlig — APPROVED** [review]
Good to go from my side once the comments are addressed.

File: doc/functions.xml (hunk 2 above)
Comment (Profpatsch) [reply doc/functions.xml]: Ah, is that confusing? I added the `{}` to make it easier to see that it’s an attrset, which is of course not the formally correct way. If it tripped you up, I think I’ll revert that.

File: lib/generators.nix (hunk 1 above)
Comment (Profpatsch) [reply lib/generators.nix]: This was a leftover, because I first thought quoting the string might be a good idea.

**Pull request merged**